cdap-docs/reference-manual/source/release-notes.rst

```diff
@@ -23,6 +23,281 @@ Cask Data Application Platform Release Notes
+  on a different architecture since we expect users to use our distributions, packages or sdk this might
+  not be an issue so closing this.
+
+- `CDAP-4133 <https://issues.cask.co/browse/CDAP-4133>`__ -
+  Added ability to get the live-info for the AppFabric system service.
+
+- `CDAP-4233 <https://issues.cask.co/browse/CDAP-4233>`__ -
+  Messages logged by Hydrator are now prefixed with the name of the stage that logged them.
+
+- `CDAP-4453 <https://issues.cask.co/browse/CDAP-4453>`__ -
+  Added Schema validation for sources, transforms and sinks, that will validate the pipeline stages schema during
+  deployment and report issues if any.
+
+- `CDAP-4392 <https://issues.cask.co/browse/CDAP-4392>`__ -
+  Upgraded netty-http dependency in CDAP to 0.14.0
+
+- `CDAP-4536 <https://issues.cask.co/browse/CDAP-4536>`__ -
+  Master service will check that the file system, yarn, and hbase are available during startup and
+  error out if any of them are not available.
+
+- `CDAP-4519 <https://issues.cask.co/browse/CDAP-4519>`__ -
+  Master service will check file system permissions on start up and error out if they are incorrect.
+
+- `CDAP-4523 <https://issues.cask.co/browse/CDAP-4523>`__ -
+  Added the config setting master.startup.checks.enabled to control whether master startup checks are run or not
+
+- `CDAP-4518 <https://issues.cask.co/browse/CDAP-4518>`__ -
+  Master service will now log important configuration settings on startup.
+
+- `CDAP-2943 <https://issues.cask.co/browse/CDAP-2943>`__ -
+  Removed the deprecated builder-style Flow API
+
+Bug Fixes
+---------
+- `CDAP-3694<https://issues.cask.co/browse/CDAP-3694>`__ -
+  Fixed test failures in the PurchaseHistory, StreamConversion, and WikipediaPipeline example apps included in the sdk.
+
+- `CDAP-4240<https://issues.cask.co/browse/CDAP-4240>`__ -
+  Fixed a bug in the CLI command to load an artifact, where the wrong artifact name and version is
+  used if the artifact name ends with a number.
+
+- `CDAP-4532<https://issues.cask.co/browse/CDAP-4532>`__ -
+  Fixed checkstyle errors on windows
+
+- `CDAP-4294<https://issues.cask.co/browse/CDAP-4294>`__ -
+  Fixed a bug where plugins from another namespace were visible when creating an application using a system artifact.
+
+- `CDAP-4092<https://issues.cask.co/browse/CDAP-4092>`__ -
+  Fixes an issue that causes worker threads to go into an infinite recursion while exceptions
+  are being thrown in channel handlers.
+
+- `CDAP-3924<https://issues.cask.co/browse/CDAP-3924>`__ -
+  Fix QueryClient to work against HTTPS.
+
+- `CDAP-3742<https://issues.cask.co/browse/CDAP-3742>`__ -
+  Fix a bug where certain MapReduce metrics were not being properly emitted when using multiple outputs.
+
+- `CDAP-4067<https://issues.cask.co/browse/CDAP-4067>`__ -
+  Fixed an issue where socket connections to the TransactionManager were not being closed.
+
+- `CDAP-3887<https://issues.cask.co/browse/CDAP-3887>`__ -
+  CDAP Router times out idle connections
```

Comment:
it would be helpful to add a bit more context here - idle for how long? based on configuration?